src/commands/setup.ts

```diff
@@ -245,6 +247,88 @@ const error = (interaction: ChatInputCommandInteraction, content: string) => {
         });
       },
     },
+    {
+      name: SubcommandName.Ticket,
+      chatInputRun: async (interaction: ChatInputCommandInteraction) => {
+        await interaction.deferReply({ephemeral: true});
+
+        if (!interaction.inGuild()) {
+          await error(interaction, 'Command only available in servers');
+          return;
+        }
+
+        const guild = await interaction.client.guilds.fetch(
+          interaction.guildId
+        );
+        if (!guild.members.me) {
+          await error(interaction, 'I am not a member of this server');
+          return;
+        }
+
+        const ticketChannel = await guild.channels.fetch(
+          interaction.options.getChannel(TicketOption.Channel, true).id
+        );
+
+        if (!ticketChannel) {
+          await error(interaction, `Could not find ${ticketChannel} in this server`);
+          return;
+        }
+        if (!ticketChannel.isTextBased()) {
+          await error(interaction, `${ticketChannel} is not a text channel`);
+          return;
+        }
+
+        const missingChannelPermissions = guild.members.me
+          .permissionsIn(ticketChannel)
+          .missing([
+            PermissionsBitField.Flags.ViewChannel,
+            PermissionsBitField.Flags.SendMessages,
+            PermissionsBitField.Flags.EmbedLinks,
+            PermissionsBitField.Flags.AttachFiles,
+            PermissionsBitField.Flags.ManageChannels,
+            PermissionsBitField.Flags.ManageThreads,
+          ]);
+        if (missingChannelPermissions.length) {
+          await error(
+            interaction,
+            `I am missing the following permissions in ${ticketChannel}: ${missingChannelPermissions}`
+          );
+          return;
+        }
+
+        await settingsManager.set(guild.id, {
+          ticketChannel: ticketChannel.id
+        });
+
+        await ticketChannel.send({
+          embeds: [
+            new EmbedBuilder()
+              .setColor(Color.Yellow)
+              .setTitle('Staff Tickets')
+              .setDescription(
+                'Speak privately with the staff. Press the button below to start!'
+              ),
+          ],
+          components: [
+            new ActionRowBuilder<ButtonBuilder>().setComponents(
+              new ButtonBuilder()
+                .setCustomId(TButtonId.Ticket)
+                .setStyle(ButtonStyle.Primary)
+                .setLabel('🎫 Create a Ticket')
```

Comment:
What do you think about using `setEmoji` instead of embedding the emoji in the label?
```suggestion
                .setEmoji('🎫')
                .setLabel('Create a Ticket')
```